language/evm/move-to-yul/src/functions.rs

```diff
@@ -105,30 +106,119 @@ impl<'a> FunctionGenerator<'a> {
             // Compute control flow graph, entry block, and label map
             let code = target.data.code.as_slice();
             let cfg = StacklessControlFlowGraph::new_forward(code);
-            let entry_bb = Self::get_actual_entry_block(&cfg);
             let label_map = Self::compute_label_map(&cfg, code);
 
-            // Emit state machine to represent control flow.
-            // TODO: Eliminate the need for this, see also
-            //    https://medium.com/leaningtech/solving-the-structured-control-flow-problem-once-and-for-all-5123117b1ee2
-            if cfg.successors(entry_bb).iter().all(|b| cfg.is_dummmy(*b)) {
-                // In this trivial case, we have only one block and can omit the state machine
-                if let BlockContent::Basic { lower, upper } = cfg.content(entry_bb) {
-                    for offs in *lower..*upper + 1 {
-                        self.bytecode(ctx, fun_id, target, &label_map, &code[offs as usize], false);
+            if !ctx.options.apply_cfg_to_scf() {
```

Comment:
This code has become too large. Can we factor the blocks in the if/else here out into a helper functions?.